lib/Adapter/Tolerant/Indexer/MemberIndexer.php

```diff
@@ -0,0 +1,149 @@
+
+        if (empty($memberName)) {
+            return;
+        }
+
+        $memberType = $this->resolveScopedPropertyAccessMemberType($node);
+
+        $this->writeIndex($index, $memberType, $containerFqn, $memberName, $info, $node);
+    }
+
+    public function beforeParse(Index $index, SplFileInfo $info): void
+    {
+        $fileRecord = $index->get(FileRecord::fromFileInfo($info));
+        assert($fileRecord instanceof FileRecord);
+
+        foreach ($fileRecord->references() as $outgoingReference) {
+            if ($outgoingReference->type() !== MemberRecord::RECORD_TYPE) {
+                continue;
+            }
+
+            $record = $index->get(MemberRecord::fromIdentifier($outgoingReference->identifier()));
+            assert($record instanceof MemberRecord);
+            $record->removeReference($fileRecord->identifier());
+            $index->write($record);
+        }
+    }
+
+    private function resolveScopedPropertyAccessMemberType(ScopedPropertyAccessExpression $node): string
+    {
+        if ($node->parent instanceof CallExpression) {
+            return 'method';
+        }
+
+        if ($node->memberName instanceof Variable) {
+            return 'property';
+        }
+
+        if ($node->parent instanceof ExpressionStatement) {
+            return 'constant';
+        }
+
+        return 'unknown';
+    }
+
+    private function resolveMemberAccessType(MemberAccessExpression $node): string
+    {
+        if ($node->parent instanceof CallExpression) {
+            return 'method';
+        }
+
+        return 'property';
+    }
+
+    private function resolveScopedPropertyAccessName(ScopedPropertyAccessExpression $node): string
+    {
+        $memberName = $node->memberName;
+
+        if ($memberName instanceof Token) {
+            return (string)$memberName->getText($node->getFileContents());
+        }
+
+        return $memberName->getText();
+    }
+
+    private function indexMemberAccessExpression(Index $index, SplFileInfo $info, MemberAccessExpression $node): void
+    {
+        $memberName = $node->memberName;
+
+        /** @phpstan-ignore-next-line Member name could be NULL */
+        if (null === $memberName) {
+            return;
+        }
+
+        $memberName = $memberName->getText($node->getFileContents());
+
+        if (empty($memberName)) {
+            return;
+        }
+
+        $memberType = $this->resolveMemberAccessType($node);
+
+        $this->writeIndex($index, $memberType, null, (string)$memberName, $info, $node);
+    }
+
+    private function writeIndex(Index $index, string $memberType, ?string $containerFqn, string $memberName, SplFileInfo $info, Node $node): void
+    {
+        $record = $index->get(MemberRecord::fromMemberReference(MemberReference::create($memberType, $containerFqn, $memberName)));
+        assert($record instanceof MemberRecord);
+        $record->setLastModified($info->getCTime());
+        $record->setFilePath($info->getPathname());
```

Comment:
This is incorrect -- member references have no "path" as they aggregate paths to where they are referenced.